Cargo.toml

```diff
@@ -6,6 +6,12 @@ members = [
     "font",
     "winpty"
 ]
+default-members = [
+    "alacritty",
+    "alacritty_terminal",
+    "copypasta",
+    "font"
+]
```

Comment:
Now this is an interesting change. Based on these changes I'd assume that the fundamental idea is that we do not want to build anything unless we actually need it. Even if something is not part of the workspace, the dependency will be pulled in, so theoretically we'd only need `alacritty` as the only `default-member`.

However there are some things that do not 'just work'. While `cargo build` resolves all dependencies through their respective `Cargo.toml` files and `cargo clippy` and `cargo fmt` also always run on the complete workspace, regardless of `default-member` configuration (though I'm not sure if it will stay this way), the `cargo test` command will always only run for the crates specified as `default-members`.

The `default-members` behavior can be overwritten though, by running `cargo test --all`. However the `members` array can also be overwritten by running `cargo test -p <crate>`.

Since all depedencies on the `winpty` crate are feature-gated behind `cfg(windows)` annotations though, the build of `cargo build -p winpty` takes only 0.25s. So as far as I can tell this does not improve build times at all.

So is there any point in excluding winpty here?

Reply:
At present it only makes the non-windows build plan more sane, and changes nothing about the default windows build plan. After the winpty package is put behind a default feature, a windows no-default-features build plan will enjoy similar sanity.

It is might not save much in build time, but it does save a lot of time for those who are trying to clean up the dependencies. It is a nice to have, not a need to have - so I'll drop it if you objection to its inclusion.

Reply:
> and changes nothing about the default windows build plan

It will affect the default windows build plan by not running winpty tests anymore.

> It is might not save much in build time, but it does save a lot of time for those who are trying to clean up the dependencies

It does not save any depedencies, all depedencies on all platforms are always part of the lockfile. The depedencies are never installed on non-windows platforms, regardless of inclusion in the defaults or not.

> After the winpty package is put behind a default feature, a windows no-default-features build plan will enjoy similar sanity.

This could still be done once the change is actually made, so I don't see any advantage in doing it now. Though if you want, we could add the WinPTY flag in this feature too, then there might be a bit more substance and reason for the changes made.